components/micro-gateway-core/src/main/ballerina/src/gateway/handlers/jwt_auth_handler.bal

```diff
@@ -62,22 +151,193 @@ public type JWTAuthHandler object {
         string credential = headerValue.substring(6, headerValue.length()).trim();
         var authenticationResult = self.jwtAuthProvider.authenticate(credential);
         if (authenticationResult is boolean) {
-            setBackendJwtHeader(credential, req);
+            boolean generationStatus = generateAndSetBackendJwtHeader(credential,
+                                                                        req,
+                                                                        self.enabledJWTGenerator,
+                                                                        self.classLoaded,
+                                                                        self.skewTime,
+                                                                        self.enabledCaching);
+            if (!generationStatus) {
+                printError(KEY_JWT_AUTH_PROVIDER, "JWT Generation failed");
+            }
             return authenticationResult;
         } else {
             return prepareAuthenticationError("Failed to authenticate with jwt bearer auth handler.", authenticationResult);
         }
     }
+
 };
 
-public function setBackendJwtHeader(string credential, http:Request req) {
-    (jwt:JwtPayload | error) payload = getDecodedJWTPayload(credential);
-    if (payload is jwt:JwtPayload) {
-        map<json>? customClaims = payload?.customClaims;
-        // validate backend jwt claim and set it to jwt header
-        if (customClaims is map<json> && customClaims.hasKey(BACKEND_JWT)) {
-            printDebug(KEY_JWT_AUTH_PROVIDER, "Set backend jwt header.");
-            req.setHeader(jwtheaderName, customClaims.get(BACKEND_JWT).toString());
+# Identify the api details from the subscribed apis in the authentication token.
+#
+# + payload - The payload of the authentication token
+# + apiName - name of the current API
+# + apiVersion - version of the current API
+# + return - Returns map<string> with the extracted details.
+public function getAPIDetails(jwt:JwtPayload payload, string apiName, string apiVersion) returns map<string> {
```

Comment:
I remember this code is duplicated in a similar util method too. If so, can we check whether to combine them?

Reply:
@AmaliMatharaarachchi yes. In subscription validation what it does is, it adds some of the details to the auth context. but some of them are not added. I didn't changed it since I have to change the auth context record.

Reply:
Can't we move the value reading from token to a util method and use it for both subscription method and this? to avoid decoding the token multiple times.